source/base/mpi.cc

```diff
@@ -1033,6 +1033,84 @@ namespace Utilities
     }
 
 
+    namespace internal
+    {
+      template <typename T>
+      std::enable_if_t<is_mpi_type<T>(), T>
+      builtin_single_broadcast(const MPI_Comm &   comm,
+                               T                  object_to_send,
+                               const unsigned int root_process)
+      {
+#ifndef DEAL_II_WITH_MPI
+        (void)comm;
+        (void)root_process;
+        return object_to_send;
+#else
+        MPI_Datatype datatype =
+          dealii::Utilities::MPI::internal::mpi_type_id(&object_to_send);
+
+        int ierr = MPI_Bcast(&object_to_send, 1, datatype, root_process, comm);
+        AssertThrowMPI(ierr);
+
+        return object_to_send;
+#endif
+      }
+
+
+      template bool
+      builtin_single_broadcast(const MPI_Comm &, bool, const unsigned int);
```

Comment:
I think you need to write this as follows for it to work:
```suggestion
      template std::enable_if_t<true,bool>
      builtin_single_broadcast(const MPI_Comm &, bool, const unsigned int);
```
Same for the other places below of course.

Reply:
Alternatively, we could also just use `static_assert` to check that the type is OK in this function.

Reply:
@bangerth Could you explain me what's the problem and why we fix it like that? It looks to me that the two templates are almost equivalent.